src/safeposix/syscalls/net_calls.rs

```diff
@@ -1234,4 +1233,274 @@ impl Cage {
+
+        let mut return_code: i32 = 0;
+        let start_time = interface::starttimer();
+
+        let end_time = match timeout {
+            Some(time) => time,
+            None => interface::RustDuration::MAX
+        };
+
+        loop {
+            for structpoll in &mut *fds {
+                let fd = structpoll.fd;
+                let events = structpoll.events;
+
+                let read = if events & POLLIN > 0 {true} else {false};
+                let write = if events & POLLOUT > 0 {true} else {false};
+                let err = if events & POLLERR > 0 {true} else {false};
+
+                let mut reads = interface::RustHashSet::<i32>::new();
+                let mut writes = interface::RustHashSet::<i32>::new();
+                let mut errors = interface::RustHashSet::<i32>::new();
+
+                let mut mask: u32 = 0;
+
+                //passing None does the same thing as passing 0 in native Linux
+                //which essentially sets the timeout to the max value allowed (which is almost always more than enough time)
+                if Self::select_syscall(&self, fd, &mut reads, &mut writes, &mut errors, None) > 0 {
+                    mask += if !reads.is_empty() {POLLIN} else {0};
+                    mask += if !writes.is_empty() {POLLOUT} else {0};
+                    mask += if !errors.is_empty() {POLLERR} else {0};
+                    return_code += 1;
+                }
+                structpoll.revents = mask;
+            }
+
+            if return_code != 0 || interface::readtimer(start_time) > end_time {
+                break;
+            } else {
+                interface::sleep(interface::RustDuration::MILLISECOND);
+            }
+        }
+        return return_code;
+    }
+
+    pub fn _epoll_object_allocator(&self) -> i32 {
+
+        //seems to only be called in functions that don't have a filedesctable lock, so not passing the lock.
+        let mut fdtable = self.filedescriptortable.write().unwrap();
+        
+        //get a file descriptor
+        if let Some(newfd) = self.get_next_fd(None, None) {
+            //new epoll fd
+            let epollobjfd = EpollDesc {
+                mode: 0000,
+                registered_fds: interface::RustHashMap::<i32, EpollEvent>::new(),
+                advlock: interface::AdvisoryLock::new(),
+                errno: 0,
+                flags: 0
+            };
+            
+            //insert into the fdtable and return the fd i32
+            let wrappedsock = interface::RustRfc::new(interface::RustLock::new(Epoll(epollobjfd)));
+            fdtable.insert(newfd, wrappedsock);
+            return newfd;
+        } else {
+            return syscall_error(Errno::ENFILE, "epoll create", "no available file descriptor number could be found");
+        }
+    }
+
+    pub fn _epoll_object_deallocator(&self) -> i32 {
+        //seems to only pass in Repy
+        return 0;
+    }
```

Comment:
Because the object will be freed when it's dropped from the fdtable, no need to even have this function